### PR TITLE
subscription: Registration button sensitivity handling

### PR DESCRIPTION
**Pull request by M4rtinK**
The conditions for the Register/Unregister button being sensitive
are as follows:
- no registration/unregistration is currently in progress
- credentials for the currently selected authentication method are
  sufficient (we use the SECRET_TYPE_* constant to detect both locally
  set credentials as well as credentials stored in the module)

### CODE REVIEW

File: pyanaconda/ui/gui/spokes/subscription.py [hunk 1]
```diff
@@ -1008,4 +1009,24 @@ def _check_connectivity(self):
...
     def _update_register_button_state(self):
         """Update register button state."""
-        # TODO
+        if self._registration_controls_enabled:
+            # check if credentials are sufficient for registration
+            if self.authentication_method == AuthenticationMethod.USERNAME_PASSWORD:
+                username_set = self.subscription_request.account_username
+                password_type = self.subscription_request.account_password.type
+                password_set = password_type in (SECRET_TYPE_TEXT, SECRET_TYPE_HIDDEN)
+                if username_set and password_set:
```
Comment (poncovka) [inline pyanaconda/ui/gui/spokes/subscription.py:1018]: I think I have already seen these checks.

File: pyanaconda/ui/gui/spokes/subscription.py [hunk 2]
```diff
@@ -34,6 +34,7 @@
 from pyanaconda.modules.common.constants.services import SUBSCRIPTION, NETWORK
 from pyanaconda.modules.common.structures.subscription import SystemPurposeData, \
     SubscriptionRequest
+from pyanaconda.modules.common.structures.secret import SECRET_TYPE_TEXT, SECRET_TYPE_HIDDEN
```
Comment (poncovka) [inline pyanaconda/ui/gui/spokes/subscription.py:37]: These should be imported from `pyanaconda.core.constants`.

Comment (M4rtinK) [reply pyanaconda/ui/gui/spokes/subscription.py:37]: Fixed.

File: pyanaconda/ui/gui/spokes/subscription.py (hunk 1 above)
Comment (M4rtinK) [reply pyanaconda/ui/gui/spokes/subscription.py:1018]: Well we have this checks:
https://github.com/rhinstaller/anaconda/blob/master/pyanaconda/ui/lib/subscription.py#L93
But that creates a new Subscription module DBus proxy on each invocation, which I don't think is something we want to here, as this check is invoked basically on every key press on the authentication. But I guess I could add an optional `subscription_request` parameter to the checking function, that would make it skip fetching the subscription request structure over DBus.

Comment (M4rtinK) [reply pyanaconda/ui/gui/spokes/subscription.py]: OK, so with the helper function it really looks quite a bit nicer! :)

**Comment by M4rtinK** [comment]
PR updated:
- fixed `SECRET_TYPE_*` imports
- reused helper function for checking if authentication data is sufficient
- handled the subscribed state explicitly, not just as a side effect of sufficient authentication data being a pre-condition for successful subscription

**Pull request merged**